Libraries/LibWeb/Streams/AbstractOperations.cpp

```diff
@@ -289,8 +289,372 @@ bool readable_stream_has_default_reader(ReadableStream const& stream)
     return false;
 }
 
+// https://streams.spec.whatwg.org/#ref-for-in-parallel
+class ReadableStreamPipeTo final : public JS::Cell {
+    GC_CELL(ReadableStreamPipeTo, JS::Cell);
+    GC_DECLARE_ALLOCATOR(ReadableStreamPipeTo);
+
+public:
+    void process()
+    {
+        auto when_ready = GC::create_function(m_realm->heap(), [this, reacted = false](JS::Value) mutable -> WebIDL::ExceptionOr<JS::Value> {
+            if (!reacted) {
+                reacted = true;
+                read_chunk();
+            }
+
+            return JS::js_undefined();
+        });
+
+        if (auto promise = m_writer->ready())
+            WebIDL::react_to_promise(*promise, when_ready, when_ready);
```

Comment:
... or maybe i'm the silly one and the default behavior of `react_to_promise` should be to use the passed in gc functions for both resolved and rejected?

Reply:
I don't think it should be the default - it's just in this specific case, we need to run through the same steps whether the stream has become ready or erred. That is, go through the motions of checking the forward/backward close/error states of each stream in order.

Reply:
I've gone back on this though - for clarity, and so no one uses it as a bad example, I've given the error cases their own explicit check-for-shutdown callback.